internal/armada/server/job_validation.go

```diff
@@ -13,16 +15,30 @@ import (
 func validateJobsCanBeScheduled(
 	jobs []*api.Job,
 	allClusterSchedulingInfo map[string]*api.ClusterSchedulingInfoReport,
-) (bool, error) {
+) (bool, []*api.JobSubmitResponseItem, error) {
 	activeClusterSchedulingInfo := scheduling.FilterActiveClusterSchedulingInfoReports(allClusterSchedulingInfo)
+	responseItems := make([]*api.JobSubmitResponseItem, 0, len(jobs))
 	for i, job := range jobs {
 		if ok, err := scheduling.MatchSchedulingRequirementsOnAnyCluster(job, activeClusterSchedulingInfo); !ok {
 			if err != nil {
-				return false, errors.WithMessagef(err, "%d-th job can't be scheduled", i)
+				response := &api.JobSubmitResponseItem{
+					JobId: job.Id,
+					Error: errors.WithMessagef(err, "%d-th job can't be scheduled", i).Error(),
```

Comment:
I think here you're creating an error only to immediately throw it away.  It might be simpler to just use a `fmt.sprintf`

Reply:
Good catch - fixed in commit f1abe37a